File: chap11/RepertoireFichier.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let tailleRepertoire = 10
let nom = Array.create tailleRepertoire ""
let tel = Array.create tailleRepertoire ""

let _ =
  let ic = open_in "repertoire.txt" in

  for i = 0 to tailleRepertoire - 1 do
    nom.(i) <- input_line ic;
    tel.(i) <- input_line ic;
  done;
  close_in ic;

  let s = input_line stdin in
  let i = ref 0 in
  while !i < 10 && not ( s = nom.(!i) ) do
    i := !i + 1
  done;
  if !i < 10 then
   ( print_string tel.( !i) ; print_newline () )
  else
   ( print_string "Inconnu"; print_newline () )

File: chap11/RepertoireFichier.java
// Ces programmes sont sous licence CeCILL-B V1.
class RepertoireFichier {

  public static void main (String [] args) {
    String [] nom, tel;
    int i;
    String s;
    java.util.Scanner f;
    nom = new String [10];
    tel = new String [10];
    f = Isn.openIn("repertoire.txt");
    for (i = 0; i <= 9; i = i + 1) {
      nom[i] = Isn.readStringFromFile(f);
      tel[i] = Isn.readStringFromFile(f);}
    Isn.closeIn(f);
    s = Isn.readString();
    i = 0;
    while (i < 10 && !Isn.stringEqual(s,nom[i])) {
      i = i + 1;}
    if (i < 10) {
      System.out.println(tel[i]);}
    else {
      System.out.println("Inconnu");}}}
